dnf/automatic/main.py

```diff
@@ -342,6 +342,13 @@ def main(args):
 
             gpgsigcheck(base, trans.install_set)
             base.do_transaction()
+
+            # In case of no global error occurred within the transaction,
+            # we need to check state of individual transaction items.
+            for tsi in trans:
+                if tsi.state == libdnf.transaction.TransactionItemState_ERROR:
+                    raise dnf.exceptions.Error(_('Transaction failed'))
```

Comment:
This is not a mistake but an alternative. What about to modify the last live and directly return 1?

```
        logger.error(_('Error: Transaction failed'))
        return 1
```

Reply:
I was also considering this, but then I figured there was no real benefit from the change.